src/components/Card.tsx

```diff
@@ -0,0 +1,53 @@
+import React from "react"
+import Display from "./Display.tsx"
+import {CardProps} from "../interface/CardProps.ts"
+
+const Card: React.FC<CardProps> = ({cardName, cardNumber, name, cardExpireDate, onClick}) => {
+
+    const formattedExpireDate = () => {
+        if (cardExpireDate) {
+            const {month, year} = cardExpireDate
+            return month && year ? `${month} / ${year}` : `${month}${year}`
+        }
+        return ''
+    }
+    const formattedCardNumber = () => {
+        if (cardNumber) {
+            let {first, second, third, fourth} = cardNumber
+            if (first.length > 0) {
+                first = first + "-"
+            }
+            if (second.length > 0) {
+                second = second + "-"
+            }
+            const thirdSecret = third.length === 4 ? "****-" : "*".repeat(third.length)
+            const fourthSecret = fourth.length === 4 ? "****" : "*".repeat(fourth.length)
+            return `${first}${second}${thirdSecret}${fourthSecret}`
```

Comment:
a: 하이픈 추가 시 let, if 대신 함수형으로 로직을 구현하는 것도 고려해볼 수 있습니다.

```suggestion
            const maskString = (input: string) => input.replace(/./g, '*');
            const { first, second, third, fourth } = cardNumber
            
            return [first, second, maskString(third), maskString(fourth)]
                .filter((value) => value.length >= 1)
                .join('-')
```

Reply:
훨씬 깔끔해 지는군요